### PR TITLE
Texture: Add a property to avoid deformations when rotating the texture and us…

### PR DESCRIPTION
**Pull request by Popov72**
…ing non-uniform scaling.

See https://forum.babylonjs.com/t/texture-rotation-not-working-for-non-square-images/14482

PG: https://www.babylonjs-playground.com/#T32V41#22

### CODE REVIEW

File: src/Materials/Textures/texture.ts [hunk 1]
```diff
@@ -456,7 +466,11 @@ export class Texture extends BaseTexture {
             this.vScale === this._cachedVScale &&
             this.uAng === this._cachedUAng &&
             this.vAng === this._cachedVAng &&
-            this.wAng === this._cachedWAng) {
+            this.wAng === this._cachedWAng &&
+            this.uRotationCenter === this._cachedURotationCenter &&
+            this.vRotationCenter === this._cachedVRotationCenter &&
+            this.wRotationCenter === this._cachedWRotationCenter &&
+            this.homogeneousRotationInUVTransform === this._cachedHomogeneousRotationInUVTransform) {
```
Comment (deltakosh) [inline src/Materials/Textures/texture.ts:473]: I would vote for testing the rotationCenter only if homogeanous is true (that could save 3 tests for each texture matrix per frame)

Comment (Popov72) [reply src/Materials/Textures/texture.ts:473]: Those variables are also used in the "legacy" case and to me it appears it was a bug in the previous code: updating any of these values would not update the matrix whereas it should do.

**Comment by Popov72** [comment]
Regarding the build failing, it fails in the "npm install" step, I don't know what it means...

**Comment by deltakosh** [comment]
That makes sense!
The build crashing is weird..

**Comment by wtesler** [comment]
Hi, this is hellowill89 from the forum thread. 

Very nice work @Popov72 !

I roughly added your changes in this playground and confirmed the changes do produce the desired result:

https://www.babylonjs-playground.com/#T32V41#23

**Comment by Popov72** [comment]
It is showing some errors with linting and "build and unit tests" but when clicking to see the detail there's no error...

**Comment by deltakosh** [comment]
Ok let me merge :)
I'll run a nightly

**Comment by wtesler** [comment]
Thanks so much everyone. Really appreciated the quick turnaround.

**Comment by deltakosh** [comment]
always :)

**Pull request merged**